lib/instabug_http_logger.dart

```diff
@@ -1,52 +1,43 @@
-
 import 'package:instabug_flutter/NetworkLogger.dart';
 import 'package:instabug_flutter/models/network_data.dart';
 import 'package:http/http.dart' as http;
-import 'package:meta/meta.dart';
 
 class InstabugHttpLogger {
+  void onLooger(http.Response response, {DateTime? startTime}) {
+    final Map<String, dynamic> requestHeaders = <String, dynamic>{};
+    response.request!.headers.forEach((String header, dynamic value) {
+      requestHeaders[header] = value[0];
+    });
 
-  @visibleForTesting
-  @protected
-  void logHttpResponse(http.Response response, { DateTime startTime }) {
-    if (response == null) {
-      return;
-    }
+    final http.Request? request = response.request as http.Request;
 
-    http.Request request = response.request;
     if (request == null) {
       return;
     }
 
-    NetworkData networkData = NetworkData();
-
-    // fill request data
-    networkData.url = request.url.toString();
-    networkData.method = request.method;
-    if (request.body != null) {
-      networkData.requestBody = request.body;
-    }
+    final NetworkData requestData = NetworkData(
+      startTime: startTime!,
+      method: request.method,
+      url: request.url.toString(),
+      requestHeaders: requestHeaders,
+      requestBody: request.body,
+    );
 
-    request.headers.forEach((header, value){
-      networkData.requestHeaders[header] = value;
-    });
+    final DateTime endTime = DateTime.now();
 
-    // fill response data
-    networkData.status = response.statusCode;
-    response.headers.forEach((header, value){
-      networkData.responseHeaders[header] = value;
+    final Map<String, dynamic> responseHeaders = <String, dynamic>{};
+    response.headers.forEach((String header, dynamic value) {
+      responseHeaders[header] = value[0];
```

Comment:
Hi @bluego-joe ,
First of all, thanks a lot for the helpful PR comments... Second, in this part we do not only include the first character of the header. We retrieve its value from the list returned by the interceptor (it is passed as a list not a string).

For example, the content-length header is returned as follows:
content-length: [0] ... so in the above case we just get the only value in that list through list[0]

I hope this part is more clear now 🙏 ... we will also shortly introduce an update to this part to support multi-valued headers

Reply:
I think this is might be a bug then, because when I view a submitted bug on http://dashboard.instabug.com, I see the entire header field name, but only the first character of the field value. As I read the code and documentation, I believe that `response.headers` is a `Map<String, String>`, meaning of course that `value[0]` is only the first character. But I might have this wrong, and in any case I'm sure you'll find the problem. In the meantime, I look forward to the next version that includes this PR :)